### ISSUE TITLE
MKL.jl Won't Install on JuliaPro `1.4.0-1` on Windows 10

### ISSUE THREAD

**Issue by RoyiAvital** [JuliaComputing/MKL.jl#40]
I have Windows 10 with the currently latest JulaiPro (`1.4.0-1`).
When I try to install `MKL.jl` it gets stuck on the `Building` step:

```
(@v1.4) pkg> add https://github.com/JuliaComputing/MKL.jl
    Cloning git-repo `https://github.com/JuliaComputing/MKL.jl`
   Updating git-repo `https://github.com/JuliaComputing/MKL.jl`
   Updating registry at `<UserPath>.juliapro\JuliaPro_v1.4.0-1\registries\JuliaPro`
  Resolving package versions...
  Installed PackageCompiler ─ v1.0.2
   Updating `<UserPath>.juliapro\JuliaPro_v1.4.0-1\environments\v1.4\Project.toml`
  [33e6dc65] + MKL v0.2.0 #master (https://github.com/JuliaComputing/MKL.jl)
   Updating `<UserPath>.juliapro\JuliaPro_v1.4.0-1\environments\v1.4\Manifest.toml`
  [33e6dc65] + MKL v0.2.0 #master (https://github.com/JuliaComputing/MKL.jl)
  [9b87118b] + PackageCompiler v1.0.2
   Building MKL → `<UserPath>.juliapro\JuliaPro_v1.4.0-1\packages\MKL\1edTo\deps\build.log`
```
In the above I changed the user path to a space holder `<UserPath>`.

I looked into the `build.log` file, yet it is empty.

Any idea?

**Comment by KristofferC** [comment]
It is likely not stuck. It just takes a long time. You can try 

```
Pkg.build("MKL"; verbose=true)
```

and get some output.

**Comment by RoyiAvital** [comment]
I have waited 16 hours before stopping it and reporting here.
On a fairly string PC with Fiber Optics Internet Connection.

**Comment by RoyiAvital** [comment]
By the way I haven't got to write the build command.
It seemed it started the building process automatically as part of the `add` command.

I can try `Pkg.add("MKL"; verbose=true)`, Will that work?

**Comment by KristofferC** [comment]
16 hours seem indeed too long heh. 



>  Will that work?

Maybe not but it might print something useful.

**Comment by RoyiAvital** [comment]
I tried:

```
(@v1.4) pkg> add("MKL"; verbose=true)
ERROR: Could not determine command
```

Any way to download without the build process and then start the build with verbose mode?

**Comment by andreasnoack** [comment]
You shouldn't use the package mode of the REPL. Just the `Pkg.add` function.

**Comment by RoyiAvital** [comment]
I thought it would be equivalent.
Anyway, here it is:

```
julia> Pkg.add("MKL"; verbose=true)
ERROR: type Context has no field verbose
Stacktrace:
 [1] #Context!#53 at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\Types.jl:380 [inlined]
 [2] add(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}; preserve::Pkg.Types.PreserveLevel, platform::Pkg.BinaryPlatforms.Windows, kwargs::Base.Iterators.Pairs{Symbol,Bool,Tuple{Symbol},NamedTuple{(:verbose,),Tuple{Bool}}}) at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:113
 [3] #add#27 at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:109 [inlined]
 [4] #add#24 at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:107 [inlined]
 [5] add(::String; kwargs::Base.Iterators.Pairs{Symbol,Bool,Tuple{Symbol},NamedTuple{(:verbose,),Tuple{Bool}}}) at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:106
 [6] top-level scope at none:0
```

**Comment by KristofferC** [comment]
I wrote `Pkg.build` not `Pkg.add`.

**Comment by pfitzseb** [comment]
Works fine for me, fwiw. I just get an error when trying to replace the sysimg.

**Comment by RoyiAvital** [comment]
> I wrote `Pkg.build` not `Pkg.add`.

Before doing even `add`?
Otherwise when I do `add` it automatically does the build (See my first post).

**Comment by RoyiAvital** [comment]
I tried . Here is the result:

```
julia> Pkg.build("MKL"; verbose=true)
ERROR: The following package names could not be resolved:
 * MKL (not found in project or manifest)

Stacktrace:
 [1] pkgerror(::String) at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\Types.jl:53
 [2] ensure_resolved(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}; registry::Bool) at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\Types.jl:794
 [3] ensure_resolved at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\Types.jl:759 [inlined]
 [4] build(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}; verbose::Bool, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:697
 [5] #build#108 at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:680 [inlined]
 [6] #build#105 at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:677 [inlined]
 [7] top-level scope at none:0
```

I think I understand the problem.
For some reason the process of downloading and building `MKL.jl` doesn't respect local System Environment Variables.

For instance:

```
julia> ENV["APPDATA"]
"D:\\Applications\\Programming\\JuliaPro-1.4.0-1Mkl\\Profile\\AppData"

julia> ENV["USERPROFILE"]
"D:\\Applications\\Programming\\JuliaPro-1.4.0-1Mkl\\Profile"

julia> ENV["HOMEPATH"]
"D:\\Applications\\Programming\\JuliaPro-1.4.0-1Mkl\\Profile"
```

While in https://github.com/JuliaComputing/MKL.jl/issues/40#issuecomment-605951816 you can see it resolves the paths using other mechanism with different values.

In JuliaPro 1.2.x it worked perfectly and resolved the paths according to the environment variables.

**Comment by KristofferC** [comment]
>  Otherwise when I do add it automatically does the build (See my first post).

You had already done the `add` in the first step, no?

**Comment by RoyiAvital** [comment]
See my post https://github.com/JuliaComputing/MKL.jl/issues/40#issuecomment-605980980.
I tried it all.

I think something about the paths doesn't work.
It doesn't take the path according to the environment variables.

**Comment by RoyiAvital** [comment]
@KristofferC , Any idea how to solve this path issues?

I really need MKL and I can't get it to work.

**Comment by aminya** [comment]
Did you try?:
```jl
julia>]add https://github.com/JuliaComputing/MKL.jl
julia> using Pkg; Pkg.build("MKL"; verbose=true)
```

**Comment by RoyiAvital** [comment]
I tried everything I was told.
I even wrote what the problem is. There is no need to guess.

The installation doesn't honor paths by environment variables.
I'm not sure how it resolves the paths bu as I showed in https://github.com/JuliaComputing/MKL.jl/issues/40#issuecomment-605980980 it certainly doens't use the variables as it should.

**Comment by aminya** [comment]
what is the output of the following?
```
julia> using Pkg

julia> Pkg.add(PackageSpec(url="https://github.com/JuliaComputing/MKL.jl", rev="master"))
```

**Comment by RoyiAvital** [comment]
@aminya , I don't know. I guess it works as @pfitzseb suggested.
Read the thread, all data is here.

I think only @KristofferC can say why the build process use paths different than what set in the environment variables.

**Comment by andreasnoack** [comment]
> Read the thread, all data is here.

I don't think it's clear to any of us. It would be helpful if you could just run the commands exactly as posted and provide the output exactly as printed.

**Comment by RoyiAvital** [comment]
@andreasnoack , I posted it in https://github.com/JuliaComputing/MKL.jl/issues/40#issuecomment-605980980.

```
julia> Pkg.build("MKL"; verbose=true)
ERROR: The following package names could not be resolved:
 * MKL (not found in project or manifest)

Stacktrace:
 [1] pkgerror(::String) at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\Types.jl:53
 [2] ensure_resolved(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}; registry::Bool) at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\Types.jl:794
 [3] ensure_resolved at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\Types.jl:759 [inlined]
 [4] build(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}; verbose::Bool, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:697
 [5] #build#108 at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:680 [inlined]
 [6] #build#105 at C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\API.jl:677 [inlined]
 [7] top-level scope at none:0
```

As you can see, something is wrong with the paths the build process resolves.
These are the paths:

```
julia> ENV["APPDATA"]
"D:\\Applications\\Programming\\JuliaPro-1.4.0-1Mkl\\Profile\\AppData"

julia> ENV["USERPROFILE"]
"D:\\Applications\\Programming\\JuliaPro-1.4.0-1Mkl\\Profile"

julia> ENV["HOMEPATH"]
"D:\\Applications\\Programming\\JuliaPro-1.4.0-1Mkl\\Profile"
```

In the 1st post I wrote the output to the `add` command:

```
(@v1.4) pkg> add https://github.com/JuliaComputing/MKL.jl
    Cloning git-repo `https://github.com/JuliaComputing/MKL.jl`
   Updating git-repo `https://github.com/JuliaComputing/MKL.jl`
   Updating registry at `<UserPath>.juliapro\JuliaPro_v1.4.0-1\registries\JuliaPro`
  Resolving package versions...
  Installed PackageCompiler ─ v1.0.2
   Updating `<UserPath>.juliapro\JuliaPro_v1.4.0-1\environments\v1.4\Project.toml`
  [33e6dc65] + MKL v0.2.0 #master (https://github.com/JuliaComputing/MKL.jl)
   Updating `<UserPath>.juliapro\JuliaPro_v1.4.0-1\environments\v1.4\Manifest.toml`
  [33e6dc65] + MKL v0.2.0 #master (https://github.com/JuliaComputing/MKL.jl)
  [9b87118b] + PackageCompiler v1.0.2
   Building MKL → `<UserPath>.juliapro\JuliaPro_v1.4.0-1\packages\MKL\1edTo\deps\build.log`
```

Can you tell me what's missing?

**Comment by KristofferC** [comment]
> As you can see, something is wrong with the paths the build process resolves.

Nothing is wrong with the paths. It is just a stacktrace to the error and it will have paths that are pointing to where the Julia executable is.
Again, you should run the `Pkg.build("MKL"; verbose=true)` after you have tried to run `Pkg.add`.

**Comment by RoyiAvital** [comment]
There you go:

```
Pkg.add(PackageSpec(url="https://github.com/JuliaComputing/MKL.jl", rev="master"))
    Cloning git-repo `https://github.com/JuliaComputing/MKL.jl`
   Updating git-repo `https://github.com/JuliaComputing/MKL.jl`
  Resolving package versions...
  Installed PackageCompiler ─ v1.0.2
   Updating `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\environments\v1.4\Project.toml`
  [33e6dc65] + MKL v0.2.0 #master (https://github.com/JuliaComputing/MKL.jl)
   Updating `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\environments\v1.4\Manifest.toml`
  [33e6dc65] + MKL v0.2.0 #master (https://github.com/JuliaComputing/MKL.jl)
  [9b87118b] + PackageCompiler v1.0.2
   Building MKL → `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\1edTo\deps\build.log`
```

If I get it correctly, the build phase happens in the `add` phase (At least this is what I understand from `Building MKL →`).

Anyway, it gets stuck there for hours with nothing happening and `build.txt` is empty. The previous time I waited 16 hours with nothing happening. This time I waited few mintues but since nothing happens in the `Task Manager` (No download, no CPU) I stopped.

I did what @KristofferC asked after (Well, what I also did in https://github.com/JuliaComputing/MKL.jl/issues/40#issuecomment-605980980) which resulted in:


```
julia> Pkg.build("MKL"; verbose=true)
   Building MKL → `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\1edTo\deps\build.log`
[ Info: Splicing in code to load MKL in D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Julia-1.4.0\bin\..\share\julia\base\sysimg.jl[ Info: getting precompile script from: https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl
[ Info: PackageCompiler: creating base system image (incremental=false)...
[ Info: PackageCompiler: creating system image object file, this might take a while...
Generating precompile statements...
```

It is like that for 1.5 hours.
The CPU utilization of the Julai process is 0.
System is ~8% (I can't say if it is Julia or something else).

Does it help?

**Comment by KristofferC** [comment]
Thanks, now we are getting somewhere.

Next step, try:

```
Pkg.add(Pkg.PackageSpec(url="https://github.com/JuliaComputing/MKL.jl/", rev="kc/verbose_generate"))
```

press Ctrl+C when it is trying to build (since it will just hang) and then do

```
Pkg.build("MKL"; verbose=true)
```

and see what it says.

**Comment by RoyiAvital** [comment]
OK, this is what I get:

```
Pkg.add(Pkg.PackageSpec(url="https://github.com/JuliaComputing/MKL.jl/", rev="kc/verbose_generate"))
    Cloning git-repo `https://github.com/JuliaComputing/MKL.jl/`
   Updating git-repo `https://github.com/JuliaComputing/MKL.jl/`
  Resolving package versions...
   Updating `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\environments\v1.4\Project.toml`     
  [33e6dc65] ~ MKL v0.2.0 #master (https://github.com/JuliaComputing/MKL.jl) ⇒ v0.2.0 #kc/verbose_generate (https://github.com/JuliaComputing/MKL.jl/)
   Updating `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\environments\v1.4\Manifest.toml`    
  [33e6dc65] ~ MKL v0.2.0 #master (https://github.com/JuliaComputing/MKL.jl) ⇒ v0.2.0 #kc/verbose_generate (https://github.com/JuliaComputing/MKL.jl/)
   Building MKL → `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\cEQNh\deps\build.log`
```

Then:

```
julia> Pkg.build("MKL"; verbose=true)
   Building MKL → `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\cEQNh\deps\build.log`
[ Info: Splicing in code to load MKL in D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Julia-1.4.0\bin\..\share\julia\base\sysimg.jl[ Info: getting precompile script from: https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl
[ Info: PackageCompiler: creating base system image (incremental=false)...
```

Now, should I wait (I didn't break it yet)?

**Comment by KristofferC** [comment]
Yes, wait for a couple of minutes

**Comment by RoyiAvital** [comment]
OK, here is the rest (It generated an error by itself):

```
julia> Pkg.build("MKL"; verbose=true)
   Building MKL → `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\cEQNh\deps\build.log`
[ Info: Splicing in code to load MKL in D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Julia-1.4.0\bin\..\share\julia\base\sysimg.jl[ Info: getting precompile script from: https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl
[ Info: PackageCompiler: creating base system image (incremental=false)...
[ Info: PackageCompiler: creating system image object file, this might take a while...
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
jl_uv_writecb() ERROR: broken pipe EPIPE
uv_error at .\libuv.jl:97 [inlined]
uv_write_async at .\stream.jl:974
uv_write at .\stream.jl:924
unsafe_write at .\stream.jl:1007
write at .\strings\io.jl:183 [inlined]
print at .\strings\io.jl:185 [inlined]
#with_output_color#701 at .\util.jl:403
unknown function (ip: 0000000014A8B3D8)
with_output_color##kw at .\util.jl:383 [inlined]
#printstyled#702 at .\util.jl:417 [inlined]
printstyled##kw at .\util.jl:417
display_error at .\client.jl:102
display_error at .\client.jl:106
jl_apply at /cygdrive/d/buildbot/worker/package_win64/build/src\julia.h:1692 [inlined]
do_apply at /cygdrive/d/buildbot/worker/package_win64/build/src\builtins.c:643
jl_f__apply at /cygdrive/d/buildbot/worker/package_win64/build/src\builtins.c:657 [inlined]
jl_f__apply_latest at /cygdrive/d/buildbot/worker/package_win64/build/src\builtins.c:693
#invokelatest#1 at .\essentials.jl:712 [inlined]
invokelatest at .\essentials.jl:711 [inlined]
_start at .\client.jl:486
unknown function (ip: 00000000149D054A)
unknown function (ip: 00000000004017E1)
unknown function (ip: 0000000000401BD6)
unknown function (ip: 00000000004013DE)
unknown function (ip: 000000000040151A)
BaseThreadInitThunk at C:\WINDOWS\System32\KERNEL32.DLL (unknown line)
RtlUserThreadStart at C:\WINDOWS\SYSTEM32\ntdll.dll (unknown line)
ERROR: LoadError: failed process: Process(`'D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Julia-1.4.0\bin\julia.exe' --color=yes --startup-file=no --cpu-target=native '--sysimage=D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\AppData\Local\Temp\jl_NkoAuK\sys.ji' '--project=D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\AppData\Local\Temp\jl_zXBP9A' '--output-o=D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\AppData\Local\Temp\jl_0sxSHhufbI.o' -e 'Base.reinit_stdio()
@eval Sys BINDIR = ccall(:jl_get_julia_bindir, Any, ())::String
Base.init_load_path()
Base.init_depot_path()
# This @eval prevents symbols from being put into Main
@eval Module() begin
    PrecompileStagingArea = Module()
    for (_pkgid, _mod) in Base.loaded_modules
        if !(_pkgid.name in ("Main", "Core", "Base"))
            eval(PrecompileStagingArea, :(const $(Symbol(_mod)) = $_mod))
        end
    end
    precompile_statements = String[]
        append!(precompile_statements, readlines("D:\\Applications\\Programming\\JuliaPro-1.4.0-1MKL\\Profile\\AppData\\Local\\Temp\\jl_OeH9UeMHfF"))

    for statement in sort(precompile_statements)
        # println(statement)
        try
            Base.include_string(PrecompileStagingArea, statement)
        catch
            # See julia issue #28808
            @debug "failed to execute $statement"
        end
    end
end # module
include("D:\\Applications\\Programming\\JuliaPro-1.4.0-1MKL\\Profile\\AppData\\Local\\Temp\\jl_ma9OUhps4n")
empty!(LOAD_PATH)
empty!(DEPOT_PATH)
'`, ProcessExited(1)) [1]

Stacktrace:
 [1] pipeline_error at .\process.jl:525 [inlined]
 [2] run(::Cmd; wait::Bool) at .\process.jl:440
 [3] run at .\process.jl:438 [inlined]
 [4] create_sysimg_object_file(::String, ::Array{String,1}; project::String, base_sysimage::String, precompile_execution_file::Array{String,1}, precompile_statements_file::Array{String,1}, cpu_target::String, script::String, isapp::Bool) at D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\PackageCompiler\X6PvH\src\PackageCompiler.jl:292
 [5] create_sysimage(::Array{Symbol,1}; sysimage_path::Nothing, project::String, precompile_execution_file::Array{String,1}, precompile_statements_file::Array{String,1}, incremental::Bool, filter_stdlibs::Bool, replace_default::Bool, cpu_target::String, script::String, base_sysimage::Nothing, isapp::Bool) at D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\PackageCompiler\X6PvH\src\PackageCompiler.jl:423
 [6] top-level scope at D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\cEQNh\src\install.jl:109
 [7] eval at .\boot.jl:331 [inlined]
 [8] change_blas_library(::String) at D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\cEQNh\src\install.jl:107
 [9] enable_mkl_startup() at D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\cEQNh\src\install.jl:84
 [10] top-level scope at D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\cEQNh\deps\build.jl:5
 [11] include(::String) at .\client.jl:439
 [12] top-level scope at none:5
in expression starting at D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\cEQNh\deps\build.jl:5
┌ Error: Error building `MKL`: 
└ @ Pkg.Operations C:\Users\julia\AppData\Local\Julia-1.4.0\share\julia\stdlib\v1.4\Pkg\src\Operations.jl:892
```

**Comment by KristofferC** [comment]
Okay, that was my mistake. I updated the branch. Could you do `Pkg.update("MKL")` and re-run the `Pkg.build`.

**Comment by RoyiAvital** [comment]
Just to make things in this issue, @KristofferC updated the scripts again and we found issue with `addprocs(2)` which causes the hang.

@KristofferC , Removed it and then the build process yielded:

```
julia> Pkg.build("MKL"; verbose=true)
   Building MKL → `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\.juliapro\JuliaPro_v1.4.0-1\packages\MKL\CLbdz\deps\build.log`
[ Info: Splicing in code to load MKL in D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Julia-1.4.0\bin\..\share\julia\base\sysimg.jl[ Info: getting precompile script from: https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl
[ Info: PackageCompiler: creating base system image (incremental=false)...
[ Info: PackageCompiler: creating system image object file, this might take a while...
Generating precompile statements...               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.4.0 (2020-03-21)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> 
#### inputting statement: ####
"2+2"
####
julia> 2+2
4

julia> 
#### inputting statement: ####
"print(\"\")"
####
julia> print("")

julia> 
#### inputting statement: ####
"@time 1+1"
####
julia> @time 1+1
  0.000000 seconds
2

julia> 
#### inputting statement: ####
"; pwd"
####
shell>  pwd
ERROR: IOError: could not spawn `pwd`: no such file or directory (ENOENT)
Stacktrace:
 [1] _spawn_primitive(::String, ::Cmd, ::Array{Any,1}) at .\process.jl:99
 [2] #550 at .\process.jl:112 [inlined]
 [3] setup_stdios(::Base.var"#550#551"{Cmd}, ::Array{Any,1}) at .\process.jl:196
 [4] _spawn at .\process.jl:111 [inlined]
 [5] run(::Cmd; wait::Bool) at .\process.jl:439
 [6] run at .\process.jl:438 [inlined]
 [7] repl_cmd(::Cmd, ::REPL.Terminals.TTYTerminal) at .\client.jl:69
 [8] top-level scope at REPL[3]:0

julia> 
#### inputting statement: ####
"? reinterpret"
####
help?>  reinterpret
search: reinterpret

  reinterpret(type, A)

  Change the type-interpretation of a block of memory. For arrays, this
  constructs a view of the array with the same binary data as the given array,
  but with the specified element type. For example, reinterpret(Float32,
  UInt32(7)) interprets the 4 bytes corresponding to UInt32(7) as a Float32.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> reinterpret(Float32, UInt32(7))
  1.0f-44
  
  julia> reinterpret(Float32, UInt32[1 2 3 4 5])
  1×5 reinterpret(Float32, ::Array{UInt32,2}):
   1.0f-45  3.0f-45  4.0f-45  6.0f-45  7.0f-45

julia> 
#### inputting statement: ####
"using Ra\t\x03"
####
julia> using Random^C

julia> 

julia> 
#### inputting statement: ####
"\\alpha\t\x03"
####
julia> α^C

julia> 

julia> 
#### inputting statement: ####
"\e[200~paste here ;)\e[201~\"\x03"
####
julia> paste here ;)"^C

julia> 

julia> 
#### inputting statement: ####
"\e[A\e[B\x03"
####
julia> ^C

julia> 

julia> 
#### inputting statement: ####
"123\b\b\b\x03"
####
julia> ^C

julia> 

julia> 
#### inputting statement: ####
"\b\b\x03"
####
julia> ^C

julia> 

julia> 
#### inputting statement: ####
"f(x) = x03"
####
f
#### inputting statement: ####
"f(1,2)"
####
julia> f(x) = x03
f (generic function with 1 method)

julia> f(1,2)
ERROR: 
#### inputting statement: ####
"[][1]"
####
MethodError: no method matching f(::Int64, ::Int64)
Closest candidates are:
  f(::Any) at REPL[6]:1
Stacktrace:
 [1] top-level scope at REPL[7]:1

julia> [][1]
ERROR: BoundsError: attempt to access 0-element Array{Any,1} at index [1]
Stacktrace:
 [1] getindex(::Array{Any,1}, ::Int64) at .\array.jl:787
 [2] top-level scope at REPL[8]:1

julia> 
#### inputting statement: ####
"cd(\"complet_path\t\t\x03"
####
c
#### inputting statement: ####
"import Pkg"
####
julia> cd("complet_path^C

julia> 

julia> import Pkg

julia> 
#### inputting statement: ####
"tmp = mktempdir()"
####
tmp = mkte
#### inputting statement: ####
"cd(tmp)"
####
julia> tmp = mktempdir()
"D:\\Applications\\Programming\\JuliaPro-1.4.0-1MKL\\Profile\\AppData\\Local\\Temp\\jl_sKwspy"

julia> cd(tmp)

julia> 
#### inputting statement: ####
"empty!(DEPOT_PATH)"
####
e
#### inputting statement: ####
"pushfirst!(DEPOT_PATH, tmp)"
####
julia> empty!(DEPOT_PATH)
0-element Array{String,1}

julia> pushfirst!(DEPOT_PATH, tmp)
1-element Array{String,1}:
 "D:\\Applications\\Programming\\JuliaPro-1.4.0-1MKL\\Profile\\AppData\\Local\\Temp\\jl_sKwspy"

julia> 
#### inputting statement: ####
"# Prevent cloning the General registry by adding a fake one"
julia> open(io -> print(io, "name = \"Registry\"\nuuid = \"37c07fec-e54c-4851-934c-2e3885e4053e\"\nrepo = \"https://github.com/JuliaRegistries/Registry.git\"\n[packages]\n") ,"registries/Registry/Registry.toml", "w")

julia> touch("Project.toml")
"Project.toml"

julia> 
#### inputting statement: ####
"] activate ."
####
pkg>  activate .
 Activating environment at `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\AppData\Local\Temp\jl_sKwspy\Project.toml`

(jl_sKwspy) pkg> 
#### inputting statement: ####
"\x03"
####
(jl_sKwspy) pkg> ^C

julia> 

julia> 
#### inputting statement: ####
"Pkg.add(\"Test\") # adding an stdlib doesn't require internet access"
####
julia> Pkg.add("Test") # adding an stdlib doesn't require internet access
  Resolving package versions...
   Updating `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\AppData\Local\Temp\jl_sKwspy\Project.toml`
  [8dfed614] + Test 
   Updating `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\AppData\Local\Temp\jl_sKwspy\Manifest.toml`
  [2a0f44e3] + Base64 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [9a3f8284] + Random 
  [9e88b42a] + Serialization 
  [6462fe0b] + Sockets 
  [8dfed614] + Test 

julia> 
#### inputting statement: ####
"] add Te\t\t\x03"
####
(jl_sKwspy) pkg>  add Test^C

julia> 

julia> 
#### inputting statement: ####
"] st"
####
(jl_sKwspy) pkg>  st
Status `D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\AppData\Local\Temp\jl_sKwspy\Project.toml`
  [8dfed614] Test 

(jl_sKwspy) pkg> 
#### inputting statement: ####
"\x03"
####
(jl_sKwspy) pkg> ^C

julia> 

julia> 
#### inputting statement: ####
"# Create simple artifact, bind it, then use it:"
####
julia> # Create simple artifact, bind it, then use it:
julia> open(io -> println(io, "Pkg.Artifacts.artifact\"foo\"; Pkg.Artifacts.artifact\"foo_plat\""), "load_artifact.jl", "w")

julia> 
#### inputting statement: ####
"foo_path = include(\"load_artifact.jl\")"
####
julia> foo_path = include("load_artifact.jl")
"D:\\Applications\\Programming\\JuliaPro-1.4.0-1MKL\\Profile\\AppData\\Local\\Temp\\jl_sKwspy\\artifacts\\4d5fcadc293a348e88f777dc0920f11e7d71441c"

julia> 
#### inputting statement: ####
"rm(tmp; recursive=true)"     
####
julia> rm(tmp; recursive=true)
ERROR: SystemError (with D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\AppData\Local\Temp\jl_sKwspy): rmdir: Permission denied
Stacktrace:
 [1] systemerror(::Symbol, ::Int32; extrainfo::String) at .\error.jl:168
 [2] #systemerror#50 at .\error.jl:167 [inlined]
 [3] rm(::String; force::Bool, recursive::Bool) at .\file.jl:286
 [4] top-level scope at REPL[33]:1

julia> exit()
┌ Warning: temp cleanup
│   exception =
│    SystemError (with D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Profile\AppData\Local\Temp\jl_sKwspy): rmdir: Permission denied
│    Stacktrace:
│     [1] systemerror(::Symbol, ::Int32; extrainfo::String) at .\error.jl:168
│     [2] #systemerror#50 at .\error.jl:167 [inlined]
│     [3] rm(::String; force::Bool, recursive::Bool) at .\file.jl:286
│     [4] temp_cleanup_purge(::Bool) at .\file.jl:487
│     [5] temp_cleanup_purge() at .\file.jl:481
│     [6] _atexit() at .\initdefs.jl:316
│     [7] exit at .\initdefs.jl:28 [inlined]
│     [8] exit() at .\initdefs.jl:29
│     [9] top-level scope at REPL[34]:1
│     [10] eval(::Module, ::Any) at .\boot.jl:331
│     [11] eval_user_input(::Any, ::REPL.REPLBackend) at D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Julia-1.4.0\share\julia\stdlib\v1.4\REPL\src\REPL.jl:86
│     [12] macro expansion at D:\Applications\Programming\JuliaPro-1.4.0-1MKL\Julia-1.4.0\share\julia\stdlib\v1.4\REPL\src\REPL.jl:118 [inlined]
│     [13] (::REPL.var"#26#27"{REPL.REPLBackend})() at .\task.jl:358
└ @ Base.Filesystem file.jl:491

#### FINISHED ####
 4484 generated in  94.395000 seconds (overhead  66.220728 seconds)
Downloading artifact: x86_64-w64-mingw32
Exception calling "DownloadFile" with "2" argument(s): "The remote server returned an error: (404) Not Found."
At line:16 char:1
+ $webclient.DownloadFile("https://pkg.juliacomputing.com/artifact/572b ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException
[ Info: PackageCompiler: default sysimg replaced, restart Julia for the new sysimg to be in effect
┌ Warning: temp cleanup
│   exception = IOError: unlink: permission denied (EACCES)
└ @ Base.Filesystem file.jl:491
```

Then restarting Julia (Waiting on it to download `MKL Artifact`) 👍 

```
julia> BLAS.vendor()
ERROR: UndefVarError: BLAS not defined
Stacktrace:
 [1] top-level scope at none:0
```
```

**Comment by RoyiAvital** [comment]
Is the plan to make the fix official or is it about to be solved in another package?

By the way, is there a way to see the MKL version? Where is the source of the MKL Library is set?

**Comment by KristofferC** [comment]
The source of the MKL library is from MKL_jll, you can see which one is used with

```
(@v1.4) pkg> st -m MKL_jll
Status `~/.julia/environments/v1.4/Manifest.toml`
...
  [856f044c] MKL_jll v2020.0.166+0
...
```

so here I would use MKL 2020.

**Comment by RoyiAvital** [comment]
@KristofferC , I think there is update 1 to the 2020 release cycle.

Also, why is the OpenMP downloaded is 2018 while the MKL is 2020?
I'd think it is good practice to use a single version of Intel compiler package (ICC, OpenMP, MKL, IPP).

**Comment by RoyiAvital** [comment]
I installed Julia Pro `1.4.1.274`.
I ran `Pkg.add(Pkg.PackageSpec(url="https://github.com/JuliaComputing/MKL.jl/", rev="kc/verbose_generate"))` as instructed in https://github.com/JuliaComputing/MKL.jl/issues/40#issuecomment-607182940.

Julia fails to install `MKL.jl`.

This is the output of `Pkg.build("MKL"; verbose=true)`:

julia> Pkg.build("MKL"; verbose=true)
   Building MKL → `D:\Applications\Programming\JuliaProMKL\Profile\.juliapro\JuliaPro_v1.4.1-1\packages\MKL\CLbdz\deps\build.log`
[ Info: Splicing in code to load MKL in D:\Applications\Programming\JuliaProMKL\Julia-1.4.1\bin\..\share\julia\base\sysimg.jl
[ Info: getting precompile script from: https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl
[ Info: PackageCompiler: creating base system image (incremental=false)...
[ Info: PackageCompiler: creating system image object file, this might take a while...
Generating precompile statements...               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.4.1 (2020-04-14)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> 
#### inputting statement: ####
"2+2"
####
julia> 2+2
4

julia> 
#### inputting statement: ####
"print(\"\")"
####
julia> print("")

julia> 
#### inputting statement: ####
"@time 1+1"
####
julia> @time 1+1
  0.000000 seconds
2

julia> 
#### inputting statement: ####
"; pwd"
####
shell>  pwd
ERROR: IOError: could not spawn `pwd`: no such file or directory (ENOENT)
Stacktrace:
 [1] _spawn_primitive(::String, ::Cmd, ::Array{Any,1}) at .\process.jl:99
 [2] #550 at .\process.jl:112 [inlined]
 [3] setup_stdios(::Base.var"#550#551"{Cmd}, ::Array{Any,1}) at .\process.jl:196
 [4] _spawn at .\process.jl:111 [inlined]
 [5] run(::Cmd; wait::Bool) at .\process.jl:439
 [6] run at .\process.jl:438 [inlined]
 [7] repl_cmd(::Cmd, ::REPL.Terminals.TTYTerminal) at .\client.jl:69
 [8] top-level scope at REPL[3]:0

julia> 
#### inputting statement: ####
"? reinterpret"
####
help?>  reinterpret
search: reinterpret

  reinterpret(type, A)

  Change the type-interpretation of a block of memory. For arrays, this
  constructs a view of the array with the same binary data as the given array,
  but with the specified element type. For example, reinterpret(Float32,
  UInt32(7)) interprets the 4 bytes corresponding to UInt32(7) as a Float32.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> reinterpret(Float32, UInt32(7))
  1.0f-44
  
  julia> reinterpret(Float32, UInt32[1 2 3 4 5])
  1×5 reinterpret(Float32, ::Array{UInt32,2}):
   1.0f-45  3.0f-45  4.0f-45  6.0f-45  7.0f-45

julia> 
#### inputting statement: ####
"using Ra\t\x03"
####
julia> using Random^C

julia> 

julia> 
#### inputting statement: ####
"\\alpha\t\x03"
####
julia> α^C

julia> 

julia> 
#### inputting statement: ####
"\e[200~paste here ;)\e[201~\"\x03"
####
julia> paste here ;)"^C

julia> 

julia> 
#### inputting statement: ####
"\e[A\e[B\x03"
####
julia> ^C

julia> 

julia> 
#### inputting statement: ####
"123\b\b\b\x03"
####
julia> ^C

julia> 

julia> 
#### inputting statement: ####
"\b\b\x03"
####
julia> ^C

julia> 

julia> 
#### inputting statement: ####
"f(x) = x03"
####
f
#### inputting statement: ####
"f(1,2)"
####
julia> f(x) = x03
f (generic function with 1 method)

julia> f(1,2)
ERROR: 
#### inputting statement: ####
"[][1]"
####
MethodError: no method matching f(::Int64, ::Int64)
Closest candidates are:
  f(::Any) at REPL[6]:1
Stacktrace:
 [1] top-level scope at REPL[7]:1

julia> [][1]
ERROR: BoundsError: attempt to access 0-element Array{Any,1} at index [1]
Stacktrace:
 [1] getindex(::Array{Any,1}, ::Int64) at .\array.jl:788
 [2] top-level scope at REPL[8]:1

julia> 
#### inputting statement: ####
"cd(\"complet_path\t\t\x03"
####
julia> cd("complet_path^C

julia> 

julia> 
#### inputting statement: ####
"import Pkg"
####
julia> import Pkg

julia> 
#### inputting statement: ####
"tmp = mktempdir()"
####
julia> 
#### inputting statement: ####
"cd(tmp)"
####
julia> tmp = mktempdir()
"C:\\Users\\ROYIAV~1\\AppData\\Local\\Temp\\jl_zu8LQC"

julia> cd(tmp)

julia> 
#### inputting statement: ####
"empty!(DEPOT_PATH)"
####
e
#### inputting statement: ####
"pushfirst!(DEPOT_PATH, tmp)"
####
julia> empty!(DEPOT_PATH)
0-element Array{String,1}

julia> pushfirst!(DEPOT_PATH, tmp)
1-element Array{String,1}:
 "C:\\Users\\ROYIAV~1\\AppData\\Local\\Temp\\jl_zu8LQC"

julia> 
#### inputting statement: ####
"# Prevent cloning the General registry by adding a fake one"
julia> open(io -> print(io, "name = \"Registry\"\nuuid = \"37c07fec-e54c-4851-934c-2e3885e4053e\"\nrepo = \"https://github.com/JuliaRegistries/Registry.git\"\n[packages]\n") ,"registries/Registry/Registry.toml", "w")
julia> touch("Project.toml")
"Project.toml"

julia> 
#### inputting statement: ####
"] activate ."
####

#### inputting statement: ####
"\x03"
####
pkg>  activate .
 Activating environment at `C:\Users\ROYIAV~1\AppData\Local\Temp\jl_zu8LQC\Project.toml`

(jl_zu8LQC) pkg> ^C

julia> 

julia> 
#### inputting statement: ####
"Pkg.add(\"Test\") # adding an stdlib doesn't require internet access"
####
P
#### inputting statement: ####
"] add Te\t\t\x03"
####
julia> Pkg.add("Test") # adding an stdlib doesn't require internet access
  Resolving package versions...
   Updating `C:\Users\Royi Avital\AppData\Local\Temp\jl_zu8LQC\Project.toml`
  [8dfed614] + Test 
   Updating `C:\Users\Royi Avital\AppData\Local\Temp\jl_zu8LQC\Manifest.toml`
  [2a0f44e3] + Base64 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [9a3f8284] + Random 
  [9e88b42a] + Serialization 
  [6462fe0b] + Sockets 
  [8dfed614] + Test 

(jl_zu8LQC) pkg>  add Test
#### inputting statement: ####
"] st"
####
(jl_zu8LQC) pkg>  add Test^C

julia> 

(jl_zu8LQC) pkg>  st

#### inputting statement: ####
"\x03"
####
Status `C:\Users\Royi Avital\AppData\Local\Temp\jl_zu8LQC\Project.toml`
  [8dfed614] Test 

(jl_zu8LQC) pkg> ^C

julia> 

julia> 
#### inputting statement: ####
"# Create simple artifact, bind it, then use it:"
####
julia> # Create simple artifact, bind it, then use it:
julia> open(io -> println(io, "Pkg.Artifacts.artifact\"foo\"; Pkg.Artifacts.artifact\"foo_plat\""), "load_artifact.jl", "w")

julia> 
#### inputting statement: ####
"foo_path = include(\"load_artifact.jl\")"
####
julia> foo_path = include("load_artifact.jl")
"C:\\Users\\ROYIAV~1\\AppData\\Local\\Temp\\jl_zu8LQC\\artifacts\\4d5fcadc293a348e88f777dc0920f11e7d71441c"

julia> 
#### inputting statement: ####
"rm(tmp; recursive=true)"     
####
julia> rm(tmp; recursive=true)
ERROR: SystemError (with C:\Users\ROYIAV~1\AppData\Local\Temp\jl_zu8LQC): rmdir: Permission denied
Stacktrace:
 [1] systemerror(::Symbol, ::Int32; extrainfo::String) at .\error.jl:168
 [2] #systemerror#50 at .\error.jl:167 [inlined]
 [3] rm(::String; force::Bool, recursive::Bool) at .\file.jl:286
 [4] top-level scope at REPL[33]:1

julia> exit()
┌ Warning: temp cleanup
│   exception =
│    SystemError (with C:\Users\ROYIAV~1\AppData\Local\Temp\jl_zu8LQC): rmdir: Permission denied
│    Stacktrace:
│     [1] systemerror(::Symbol, ::Int32; extrainfo::String) at .\error.jl:168
│     [2] #systemerror#50 at .\error.jl:167 [inlined]
│     [3] rm(::String; force::Bool, recursive::Bool) at .\file.jl:286
│     [4] temp_cleanup_purge(::Bool) at .\file.jl:487
│     [5] temp_cleanup_purge() at .\file.jl:481
│     [6] _atexit() at .\initdefs.jl:316
│     [7] exit at .\initdefs.jl:28 [inlined]
│     [8] exit() at .\initdefs.jl:29
│     [9] top-level scope at REPL[34]:1
│     [10] eval(::Module, ::Any) at .\boot.jl:331
│     [11] eval_user_input(::Any, ::REPL.REPLBackend) at D:\Applications\Programming\JuliaProMKL\Julia-1.4.1\share\julia\stdlib\v1.4\REPL\src\REPL.jl:86
│     [12] macro expansion at D:\Applications\Programming\JuliaProMKL\Julia-1.4.1\share\julia\stdlib\v1.4\REPL\src\REPL.jl:118 [inlined]
│     [13] (::REPL.var"#26#27"{REPL.REPLBackend})() at .\task.jl:358
└ @ Base.Filesystem file.jl:491

#### FINISHED ####
 4468 generated in  63.809000 seconds (overhead  45.393034 seconds)
ERROR: LoadError: IOError: unlink: permission denied (EACCES)
Stacktrace:
 [1] uv_error at .\libuv.jl:97 [inlined]
 [2] unlink(::String) at .\file.jl:888
 [3] rm(::String; force::Bool, recursive::Bool) at .\file.jl:268
 [4] rename(::String, ::String) at .\file.jl:899
 [5] #mv#13 at .\file.jl:393 [inlined]
 [6] mv at .\file.jl:392 [inlined]
 [7] move_default_sysimage_if_windows() at D:\Applications\Programming\JuliaProMKL\Profile\.juliapro\JuliaPro_v1.4.1-1\packages\PackageCompiler\D8oaG\src\PackageCompiler.jl:83
 [8] create_sysimage(::Array{Symbol,1}; sysimage_path::Nothing, project::String, precompile_execution_file::Array{String,1}, precompile_statements_file::Array{String,1}, incremental::Bool, filter_stdlibs::Bool, replace_default::Bool, cpu_target::String, script::String, base_sysimage::Nothing, isapp::Bool) at D:\Applications\Programming\JuliaProMKL\Profile\.juliapro\JuliaPro_v1.4.1-1\packages\PackageCompiler\D8oaG\src\PackageCompiler.jl:439
 [9] top-level scope at D:\Applications\Programming\JuliaProMKL\Profile\.juliapro\JuliaPro_v1.4.1-1\packages\MKL\CLbdz\src\install.jl:110
 [10] eval at .\boot.jl:331 [inlined]
 [11] change_blas_library(::String) at D:\Applications\Programming\JuliaProMKL\Profile\.juliapro\JuliaPro_v1.4.1-1\packages\MKL\CLbdz\src\install.jl:108
 [12] enable_mkl_startup() at D:\Applications\Programming\JuliaProMKL\Profile\.juliapro\JuliaPro_v1.4.1-1\packages\MKL\CLbdz\src\install.jl:85
 [13] top-level scope at D:\Applications\Programming\JuliaProMKL\Profile\.juliapro\JuliaPro_v1.4.1-1\packages\MKL\CLbdz\deps\build.jl:5
 [14] include(::String) at .\client.jl:439
 [15] top-level scope at none:5
in expression starting at D:\Applications\Programming\JuliaProMKL\Profile\.juliapro\JuliaPro_v1.4.1-1\packages\MKL\CLbdz\deps\build.jl:5
┌ Error: Error building `MKL`: 
└ @ Pkg.Operations C:\Users\julia\AppData\Local\Julia-1.4.1\share\julia\stdlib\v1.4\Pkg\src\Operations.jl:892


By the way, though it said `D:\Applications\Programming\JuliaProMKL\Profile\.juliapro\JuliaPro_v1.4.1-1\packages\MKL\CLbdz\deps\build.log` the file is empty.

**Comment by RoyiAvital** [comment]
@KristofferC , Any idea about this?
Any information missing I can add?

**Comment by neversakura** [comment]
This is related with [issue 181](https://github.com/JuliaLang/PackageCompiler.jl/issues/181) for PackageCompiler.jl. The problem seems to be that `move_default_sysimage_if_windows()` cannot move the original `sys.dll` away because some access issue. I don't have an elegant solution to this but my current fix is to precompile the new `sys.dll` to some other directory and manually replace the old one with it.

This can be done by adding `sysimage_path` to `create_sysimage` in the `install.jl` file
```julia
@eval begin
    using PackageCompiler
    PackageCompiler.create_sysimage(;
        incremental = false,
        sysimage_path = "some_new_directory\\sys.dll",
        replace_default = false,
        script = get_precompile_statments_file(),
    )
end
```

**Comment by alexhuangweijie** [comment]
i met some similar problems. can anyone help me ?

`julia> Pkg.build
build (generic function with 5 methods)

julia> Pkg.build("MKL")
   Building MKL → `C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\deps\build.log`
┌ Error: Error building `MKL`: 
│ [ Info: Splicing in code to load MKL in G:\JuliaPro-1.4.2-1\Julia-1.4.2\bin\..\share\julia\base\sysimg.jl
│ [ Info: getting precompile script from: https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl
│ ┌ Error: Download failed: curl: (6) Could not resolve host: raw.githubusercontent.com
│ └ @ Base download.jl:43
│ ERROR: LoadError: failed process: Process(`'C:\WINDOWS\System32\curl.exe' -s -S -g -L -f -o 'C:\Users\49032\AppData\Local\Temp\jl_iNuBem7FdO' https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl`, ProcessExited(6)) [6]
│ 
│ Stacktrace:
│  [1] pipeline_error at .\process.jl:525 [inlined]
│  [2] download_curl(::String, ::String, ::String) at .\download.jl:44
│  [3] download at .\download.jl:62 [inlined]
│  [4] get_precompile_statments_file() at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:75
│  [5] top-level scope at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:107
│  [6] eval at .\boot.jl:331 [inlined]
│  [7] change_blas_library(::String) at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:105
│  [8] enable_mkl_startup() at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:82
│  [9] top-level scope at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\deps\build.jl:5
│  [10] include(::String) at .\client.jl:439
│  [11] top-level scope at none:5
│ in expression starting at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\deps\build.jl:5
└ @ Pkg.Operations C:\Users\julia\AppData\Local\Julia-1.4.2\share\julia\stdlib\v1.4\Pkg\src\Operations.jl:892`

**Comment by alexhuangweijie** [comment]
i have a similar problem. i  want to  install MKL,jl with 10 and JuliaPro, but got  stuck in here.
'''
julia> Pkg.build
build (generic function with 5 methods)

julia> Pkg.build("MKL")
   Building MKL → `C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\deps\build.log`
┌ Error: Error building `MKL`: 
│ [ Info: Splicing in code to load MKL in G:\JuliaPro-1.4.2-1\Julia-1.4.2\bin\..\share\julia\base\sysimg.jl
│ [ Info: getting precompile script from: https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl
│ ┌ Error: Download failed: curl: (6) Could not resolve host: raw.githubusercontent.com
│ └ @ Base download.jl:43
│ ERROR: LoadError: failed process: Process(`'C:\WINDOWS\System32\curl.exe' -s -S -g -L -f -o 'C:\Users\49032\AppData\Local\Temp\jl_iNuBem7FdO' https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl`, ProcessExited(6)) [6]
│ 
│ Stacktrace:
│  [1] pipeline_error at .\process.jl:525 [inlined]
│  [2] download_curl(::String, ::String, ::String) at .\download.jl:44
│  [3] download at .\download.jl:62 [inlined]
│  [4] get_precompile_statments_file() at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:75
│  [5] top-level scope at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:107
│  [6] eval at .\boot.jl:331 [inlined]
│  [7] change_blas_library(::String) at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:105
│  [8] enable_mkl_startup() at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:82
│  [9] top-level scope at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\deps\build.jl:5
│  [10] include(::String) at .\client.jl:439
│  [11] top-level scope at none:5
│ in expression starting at C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\deps\build.jl:5
└ @ Pkg.Operations C:\Users\julia\AppData\Local\Julia-1.4.2\share\julia\stdlib\v1.4\Pkg\src\Operations.jl:892
'''

**Comment by alexhuangweijie** [comment]
i have a similar problem. i want to install MKL,jl with 10 and JuliaPro, but got stuck in here.
```
julia> Pkg.build
build (generic function with 5 methods)

julia> Pkg.build("MKL")
Building MKL → C:\Users\49032\.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\deps\build.log
┌ Error: Error building MKL:
│ [ Info: Splicing in code to load MKL in G:\JuliaPro-1.4.2-1\Julia-1.4.2\bin..\share\julia\base\sysimg.jl
│ [ Info: getting precompile script from: https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl
│ ┌ Error: Download failed: curl: (6) Could not resolve host: raw.githubusercontent.com
│ └ @ Base download.jl:43
│ ERROR: LoadError: failed process: Process('C:\WINDOWS\System32\curl.exe' -s -S -g -L -f -o 'C:\Users\49032\AppData\Local\Temp\jl_iNuBem7FdO' https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl, ProcessExited(6)) [6]
│
│ Stacktrace:
│ [1] pipeline_error at .\process.jl:525 [inlined]
│ [2] download_curl(::String, ::String, ::String) at .\download.jl:44
│ [3] download at .\download.jl:62 [inlined]
│ [4] get_precompile_statments_file() at C:\Users\49032.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:75
│ [5] top-level scope at C:\Users\49032.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:107
│ [6] eval at .\boot.jl:331 [inlined]
│ [7] change_blas_library(::String) at C:\Users\49032.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:105
│ [8] enable_mkl_startup() at C:\Users\49032.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\src\install.jl:82
│ [9] top-level scope at C:\Users\49032.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\deps\build.jl:5
│ [10] include(::String) at .\client.jl:439
│ [11] top-level scope at none:5
│ in expression starting at C:\Users\49032.juliapro\JuliaPro_v1.4.2-1\packages\MKL\1edTo\deps\build.jl:5
└ @ Pkg.Operations C:\Users\julia\AppData\Local\Julia-1.4.2\share\julia\stdlib\v1.4\Pkg\src\Operations.jl:892
```

**Comment by zhopan77** [comment]
I now face this similar problem. It happens only on one of my approx. 5 Windows 10 64-bit machines.

Building MKL.jl is stuck at "Generating precompile statements...".

```
julia> Pkg.build("MKL"; verbose=true)
   Building MKL → `C:\Users\zpan\.julia\packages\MKL\1edTo\deps\build.log`
[ Info: Splicing in code to load MKL in c:\julia-1.4.2\bin\..\share\julia\base\sysimg.jl
[ Info: getting precompile script from: https://raw.githubusercontent.com/JuliaLang/julia/release-1.4/contrib/generate_precompile.jl
[ Info: PackageCompiler: creating base system image (incremental=false)...
[ Info: PackageCompiler: creating system image object file, this might take a while...
Generating precompile statements...
```

I waited long (hours) after all other machines finished building. Tried open Julia as Administrator which didn't fix it. The CPU is Intel Core i7-3960X Extreme, if that makes any difference. Julia v1.4.2, MKL v0.2.0 #master.

**Comment by zhopan77** [comment]
Just want to report that the problem went away on the same machine on the newly released Julia 1.5.0.

**Comment by RoyiAvital** [comment]
@andreasnoack , @KristofferC , Did you close it because it was solved by you?
Is there a reason why Julia 1.5.xxx will solve the issue with replacing the files on the fly?

**Comment by alexhuangweijie** [comment]
no, i can not solve it. instead, i installed it with ubuntu without any
trouble.

Royi <notifications@github.com> 于2020年10月14日周三 下午1:27写道：

**Comment by andreasnoack** [comment]
https://github.com/JuliaComputing/MKL.jl/issues/40#issuecomment-668125482 suggested that this was no longer an issue. Is it still not working with JuliaPro?

**Comment by alexhuangweijie** [comment]
i have not checked it on windows ,since i can run it on my ubuntu device.

Andreas Noack <notifications@github.com> 于2020年10月19日周一 下午7:45写道：

**Comment by RoyiAvital** [comment]
@andreasnoack , I think previously @KristofferC mentioned the issue being Windows not able to replace DLL's (Or other files) while running.

Was there any change to bypass this limitation? If not, I think the issue should still be open.

**Comment by andreasnoack** [comment]
It would be great if somebody could confirm that it's actually still a problem. It looks like https://github.com/JuliaComputing/MKL.jl/issues/40#issuecomment-668125482 is based on a Windows machine.

**Comment by RoyiAvital** [comment]
@andreasnoack , I don't get it.
@KristofferC, has identified the problem.
Nothing has changed to take care of it.
Is there a reason why the Julia version should make it any different?

It used to work for me as well.
One time a success the other time a failure. It is not consistent. That's the whole issue.
